src/V3SplitVar.cpp

```diff
@@ -0,0 +1,783 @@
+//    1:2:1:2: VARREF var0 [LV]
+//    1:2: PRAGMA  <= PRAGMA comes after the initial VARREF
+//
+// To catch all VARREF, 2-pass scan is used.
+// 1st scan:Just collect VAR and PRAGMA
+// 2nd scan:Actual splitting
+
+class ScanPragmaVisitor : public AstNVisitor {
+    AstVar* m_lastVarp;  // the most recently declared variable
+    std::vector<std::pair<AstPragma*, AstVar*> > m_vars;
+    virtual void visit(AstNode* nodep) VL_OVERRIDE { iterateChildren(nodep); }
+    virtual void visit(AstModule* nodep) VL_OVERRIDE {
+        UASSERT_OBJ(nodep == NULL, nodep, "This class must be used within a module");
+    }
+    virtual void visit(AstVar* nodep) VL_OVERRIDE { m_lastVarp = nodep; }
+    virtual void visit(AstPragma* pragmap) VL_OVERRIDE {
+        if (pragmap->pragType() != AstPragmaType::SPLIT_VAR) return;  // nothing to do
+        // when m_lastVarp == NULL, the pragma is stray pragma.
+        m_vars.push_back(std::make_pair(pragmap, m_lastVarp));
+        m_lastVarp = NULL;
+    }
+    ScanPragmaVisitor()
+        : m_lastVarp(NULL) {}
+
+public:
+    static std::vector<std::pair<AstPragma*, AstVar*> > scan(AstNodeModule* modp) {
+        ScanPragmaVisitor v;
+        v.iterateChildren(modp);
+        return v.m_vars;
+    }
+};
+
+//######################################################################
+// Split Unpacked Variables
+
+class SplitUnpackedVarVisitor : public AstNVisitor {
+    AstNodeModule* m_modp;  // current module
+    int m_numSplit;         // total number of split variable
+    // key:variable to be split. value:location where the variable is referenced.
+    vl_unordered_map<AstVar*, std::vector<AstArraySel*> > m_refs;
+    typedef vl_unordered_set<AstNodeAssign*> AssignSet;
+    AssignSet m_assigns;  // at least LHS or RHS will be split
+
+    // used when traversing LHS and RHS of assignment. true if func/task is included
+    bool m_taskFuncFound;
+    // used when traversing LHS and RHS of assignment. added to this set.
+    typedef vl_unordered_set<AstVar*> TargetVarSet;
+    TargetVarSet m_foundTargetVar;
+
+    // This visitor is used before V3Const::constifyAllLint(),
+    // some parameters need to be resolved here, but don't abuse this function.
+    static AstConst* constifyIfNot(AstNode* nodep) {
+        AstConst* constp = VN_CAST(nodep, Const);
+        if (!constp) {
+            UINFO(4, nodep << " is expected to be constant, but not\n");
+            AstNode* constified = V3Const::constifyEdit(nodep);
+            UINFO(4, "After constified:" << constified << '\n');
+            constp = VN_CAST(constified, Const);
+        }
+        return constp;
+    }
+    static int outerMostSizeOfUnpackedArray(AstVar* nodep) {
+        AstUnpackArrayDType* dtypep = VN_CAST(nodep->dtypep(), UnpackArrayDType);
+        UASSERT_OBJ(dtypep, nodep, "Must be unapcked array");
+        return dtypep->msb() - dtypep->lsb() + 1;
+    }
+
+    virtual void visit(AstNode* nodep) VL_OVERRIDE { iterateChildren(nodep); }
```

Comment:
BTW if you'd like to make a separate pull request (off master) to add VL_OVERRIDE to all existing visitors that would be appreciated.